### ISSUE TITLE
Printing inside a mock block hangs/crashes Julia

### ISSUE THREAD

**Issue by christopher-dG** [christopher-dG/SimpleMock.jl#1]
```jl
julia> mock(g -> println("!"), get)
!ERROR: ^C  
^C^C^C^C^CWARNING: Force throwing a SIGINT
^C^C^C^CERROR: ^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^Czsh: segmentation fault (core dumped)  julia --project
```
It hangs forever until I send a bunch of sigints, then it crashes.

But not always:

```jl
julia> mock(print, identity)
Mock{Nothing,Symbol}(Symbol("##374"), Call[], Dict{Symbol,Any}(), nothing, Symbol("##371"))
```

```jl
julia> mock(get) do get
           Base.get()
           called_once(get)
       end
true

julia> mock(get) do get
           Base.get()
           @show called_once(get)
       end
called_once(get) = ERROR: 
```

**Comment by christopher-dG** [comment]
I would not be surprised if this is the same thing as #2, #3 and can be solved with #5. It seems that `get` is just a well-used function, especially in IO-land where `get(::IOContext)` is a thing. Leaving open though, since the effects are quite serious. 

The best way to avoid this for now is just to avoid mocking these super common functions.

**Comment by christopher-dG** [comment]
Easy to avoid now:

```jl
 mock(g -> println("!"), get; filters=[max_depth(1)])
 mock(g -> println("!"), get; filters=[excluding(Base)])
```

There's a also a mostly unrelated docs section about avoiding printing.
